### PR TITLE
IResearch view stored values optimizations

### PR DESCRIPTION
**Pull request by iurii-i-popov**
Stored values optimizations.

https://github.com/arangodb/backlog/issues/677

One time allocations.
Stored values buffer allocation.
Merge executor readers allocation.

### CODE REVIEW

**Comment by iurii-i-popov** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/9419/

File: arangod/Aql/IResearchViewExecutor.cpp [hunk 1]
```diff
@@ -731,77 +703,66 @@ bool IResearchViewExecutorBase<Impl, Traits>::writeRow(ReadContext& ctx,
 }
 
 template <typename Impl, typename Traits>
-void IResearchViewExecutorBase<Impl, Traits>::getStoredValue(
-    irs::document const& doc, std::vector<irs::bytes_ref>& storedValue, size_t index,
-    std::vector<irs::columnstore_reader::values_reader_f> const& storedValuesReaders) {
-  irs::columnstore_reader::values_reader_f reader = storedValuesReaders[index];
+void IResearchViewExecutorBase<Impl, Traits>::readStoredValues(
+    irs::document const& doc, size_t index) {
+  auto const& reader = _storedValuesReaders[index];
```
Comment (gnusi) [inline arangod/Aql/IResearchViewExecutor.cpp:708]: `assert(...)`?

**Comment by iurii-i-popov** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr-linux/11162/

**Comment by iurii-i-popov** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr-linux/11164/

**Comment by iurii-i-popov** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/9423/

**Pull request merged**